src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -431,13 +435,15 @@ private TypeSpec CreateDictionarySpec(TypeParseInfo typeParseInfo, ITypeSymbol k
                 TypeRef keyTypeRef = EnqueueTransitiveType(typeParseInfo, keyTypeSymbol, DiagnosticDescriptors.DictionaryKeyNotSupported);
                 TypeRef elementTypeRef = EnqueueTransitiveType(typeParseInfo, elementTypeSymbol, DiagnosticDescriptors.ElementTypeNotSupported);
 
+                Debug.Assert(!shouldTestCast || !type.IsValueType, "Should not test cast for value types.");
```

Comment:
>Debug.Assert(!shouldTestCast || !type.IsValueType, "Should not test cast for value types."); [](http://example.com/codeflow?start=16&length=92)

what happen when having a struct implementing `IReadonlyDictionary`?

Reply:
The `IReadonlyDictionary` behavior is only for when the actual type is `IReadonlyDictionary`, not for a type implementing `IReadonlyDictionary`.  This is what I tried to note in the comments I added.